### PR TITLE
Samsung Exynos AI LiteCore - Samsung Backend in test suite of ExecuTorch

### PR DESCRIPTION
**Pull request by Jiseong-oh**
### Summary

Release notes: TestSuite

The Samsung Exynos backend in TestSuite of ExecuTorch can be verfied and it was passed Testsuite for E9955/E9965 SoCs.

### Test plan
pytest -c /dev/nul backends/test/suite/operators/test_add.py -m backend_samsung -vs --html=./report/add_report.html
pytest -c /dev/nul backends/test/suite/models/ -m backend_samsung  --html=./report/samsung_models.html

cc @SS-JIA @digantdesai @kimishpatel

### CODE REVIEW

File: backends/samsung/builders/op_getitem.py [hunk 1]
```diff
@@ -28,5 +28,5 @@ def define_node(
         node: torch.fx.Node,
         enn_graph: EnnGraph,
         vals_to_ids: Dict[torch.Tensor, int],
-    ) -> None:
-        return
+    ) -> bool:
+        return True
```
Comment (SS-JIA) [inline backends/samsung/builders/op_getitem.py:32]: [P1] This visitor advertises support while emitting neither a tensor nor an op. Because singleton partitions are allowed, a `getitem` whose tuple producer is not delegated can become a `getitem`-only delegate and fail serialization; the new shape-only visitor has the same issue. Please treat these nodes as supported only in the context of a supported producer/consumer, or filter partitions that emit no ENN ops.

Generated by an AI reviewer.

File: backends/samsung/partition/enn_partitioner.py [hunk 2]
```diff
@@ -91,10 +99,19 @@ def generate_partitions(
         self, edge_program: torch.export.ExportedProgram
     ) -> List[Any]:
         self.op_support_checker = EnnOperatorSupport(edge_program, self.compile_specs)
-        return generate_partitions_from_list_of_nodes(
+        partition_list = generate_partitions_from_list_of_nodes(
             edge_program.graph_module,
             op_support=self.op_support_checker,
         )
+        if len(partition_list) == 1 and partition_list[0].size() == 1:
```
Comment (SS-JIA) [inline backends/samsung/partition/enn_partitioner.py:106]: [P2] This workaround only removes a singleton clone when it is the sole partition. If a graph also has another supported partition, an isolated clone partition survives; preprocessing then erases its only op and triggers the same graph error described below. Please filter every size-one clone partition, independent of total partition count.

Generated by an AI reviewer.

File: backends/samsung/partition/enn_partitioner.py [hunk 3]
```diff
@@ -70,9 +71,16 @@ def is_node_supported(self, _, node: torch.fx.Node) -> bool:
...
             return False
 
-        if node.target in SUPPORTED_OPS or node.target.__name__ in self.node_visitors:
+        if node.target in SUPPORTED_OPS:
             return True
 
+        if node.target.__name__ in self.node_visitors:
+            enn_graph = EnnGraph()
+            vals_to_ids: Dict[torch.fx.Node, int] = {}
+            return self.node_visitors[node.target.__name__].define_node(
```
Comment (SS-JIA) [inline backends/samsung/partition/enn_partitioner.py:80]: [P1] `define_node()` is being used as a support predicate here, but several visitors still signal unsupported variants by raising (for example non-last-axis `topk`, max-pool indices, and unsupported padding/layer-norm forms). Those cases now abort partitioning instead of returning `False` and falling back. Please make capability checking side-effect-free and non-throwing—ideally a separate `is_supported()` path—or consistently translate expected unsupported cases to `False`.

Generated by an AI reviewer.

File: backends/samsung/builders/op_batch_norm.py [hunk 4]
```diff
@@ -51,6 +51,12 @@ def define_node(
 
         output_id = self.define_tensor(node, enn_graph, vals_to_ids, output_idx=0)
 
+        users = list(node.users.keys())
+        if len(users) > 0 and users[0].target.__name__ == "getitem":
```
Comment (SS-JIA) [inline backends/samsung/builders/op_batch_norm.py]: [P1] This maps the first user to batch-norm output 0 without checking that user’s `getitem` index, and ignores any remaining users. If an auxiliary output is consumed first, it is silently wired to normalized data; multiple consumed outputs can leave later edges unresolved. Please inspect every `getitem`, map only index 0, and reject delegation when an unsupported auxiliary output is consumed.

Generated by an AI reviewer.

File: examples/samsung/executor_runner/enn_executor_runner.cpp [hunk 5]
```diff
@@ -304,23 +304,12 @@ int main(int argc, char** argv) {
...
-  double interval_1st_infs =
-      std::chrono::duration_cast<std::chrono::microseconds>(
-          after_exec - before_exec)
-          .count() /
-      1000.0;
-
   ET_LOG(Info, "Start inference.");
-  before_exec = std::chrono::high_resolution_clock::now();
+  auto before_exec = std::chrono::high_resolution_clock::now();
   for (int i = 0; i < FLAGS_num_executions; ++i) {
```
Comment (SS-JIA) [inline examples/samsung/executor_runner/enn_executor_runner.cpp:328]: [P2] The patch removes one extra execution because `execute()` may mutate DataReader-backed inputs, but warmups and `--num_executions > 1` still reuse those same buffers. Later timed iterations—and the saved final output—can therefore use mutated input. Please restore pristine input bytes before each execution, or disallow repeated execution for this input path.

Generated by an AI reviewer.

File: backends/test/suite/flows/samsung.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    is_per_channel: bool = True,
+    is_qat: bool = False,
+) -> TestFlow:
+    if quantize and quant_dtype is None:
+        raise RuntimeError("Quant dtype must be provided when quantize is true.")
+
+    def create_quantize_stage() -> Quantize:
+        quantizer = EnnQuantizer()
+        quantizer.setup_quant_params(quant_dtype, is_per_channel, is_qat)
+        return Quantize(quantizer=quantizer)
```
Comment (SS-JIA) [inline backends/test/suite/flows/samsung.py:27]: [P2] This constructs the generic harness `Quantize` stage, bypassing Samsung’s backend-specific stage in `samsung_tester.py`. The Samsung stage decomposes SDPA and deliberately uses `convert_pt2e(..., fold_quantize=False)`; existing Samsung quantized tests rely on that behavior. Please import/use Samsung’s `Quantize` here so `samsung_a8w8` exercises the actual backend pipeline.

Generated by an AI reviewer.

**Review by SS-JIA — APPROVED** [review]

File: backends/samsung/builders/op_getitem.py (hunk 1 above)
Comment (Jiseong-oh) [reply backends/samsung/builders/op_getitem.py:32]: I will keep this issue but it's not needed now.

File: backends/samsung/partition/enn_partitioner.py (hunk 3 above)
Comment (Jiseong-oh) [reply backends/samsung/partition/enn_partitioner.py:80]: I will keep this issue but it's not needed now.

File: examples/samsung/executor_runner/enn_executor_runner.cpp (hunk 5 above)
Comment (Jiseong-oh) [reply examples/samsung/executor_runner/enn_executor_runner.cpp:328]: fixed

File: backends/test/suite/flows/samsung.py (hunk 6 above)
Comment (Jiseong-oh) [reply backends/test/suite/flows/samsung.py:27]: fixed
